### DISCUSSION TITLE
Setting custom properties on repository creation

### DISCUSSION THREAD

**Discussion by tomkalesse** [community/community#136463] [Repositories] [answered]
### Select Topic Area

Question

### Body

I tried to allow organization members to set custom properties when creating a repository. 

As a owner I'm able set and edit properties. But as a member not. So I created a **Repository Role** with the permission "Edit values of custom properties that allow it". With that its possible for members to change the custom properties for repos after creation, but in the creation process its not possible for members to select anything else than the default value.

Is there a permission like "Set values of custom properties that allow it" which I missed or other ways to allow members to choose a different custom property than default while creating a new repo?

Thanks for help!

Background: I provide a set of branch protection rules for the orga which can be enabled or chosen for each repo via the custom property.

**Comment by GabrielAlmeidag** [comment, marked as answer]
@tomkalesse 

Currently, there isn't a specific permission like "Set values for custom properties that allow this" that would let members select custom properties different from the default value when creating a new repository. The repository role you configured with the "Edit custom property values that allow this" permission enables members to modify properties after the repository is created, but during the creation process, the available options are limited to the default value.

One possible solution could be to create an approval process or automatic adjustment after the repository is created, where the custom properties could be reviewed and adjusted as needed. This can be done through automations or scripts that check and change the properties right after the repository is created.
